src/System.IO.FileSystem/src/System/IO/Win32FileStream.cs

```diff
@@ -547,6 +562,16 @@ private void FlushWrite(bool calledFromFinalizer)
                 {
                     writeTask.GetAwaiter().GetResult();
                 }
+
+                // If not completing synchronously, ensure the async flush operation
+                // is tracked in case of incoming WriteAsync calls.
+                else
+                {
+                    if (ActiveBufferOperation())
+                        _activeBufferOperation = Task.WhenAll(_activeBufferOperation, writeTask);
+                    else
+                        _activeBufferOperation = writeTask;
```

Comment:
Every time we call FlushWriteAsync, we do these four lines to update _activeBufferOperation.  Should they just be moved into FlushWriteAsync?  This is the only place where there's the potential for avoiding it, and it's an uncommon case (an async FileStream being finalized with data in the buffer) and a cheap enough operation that the overhead doesn't seem problematic even if it's unnecessary.  Worst case, it could be moved into its own helper rather than repeating it each time.